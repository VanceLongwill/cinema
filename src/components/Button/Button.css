.Button {
  box-shadow: none;
  border: none;
  padding: var(--spacing-md);
  font-weight: 700;
  font-size: 1.5rem;
  cursor: pointer;
  border-bottom: 5px solid transparent;
  transition: 0.2ms ease-out;
}

.Button:hover {
  border-color: black;
}

.Button-positive {
  background: var(--positive-color);
}

.Button-negative {
  background: var(--negative-color);
}
