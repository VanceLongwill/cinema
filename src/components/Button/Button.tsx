import React from "react";
import cx from "classnames";

import "./Button.css";

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  actionType?: "positive" | "negative";
}

export const Button: React.FunctionComponent<ButtonProps> = ({
  actionType,
  className,
  ...props
}) => (
  <button
    className={cx(className || "", {
      Button: true,
      "Button-positive": actionType === "positive",
      "Button-negative": actionType === "negative"
    })}
    {...props}
  />
);
