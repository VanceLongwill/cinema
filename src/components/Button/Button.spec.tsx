import * as React from "react"

import { mount } from "enzyme"

import { Button } from "./Button"

describe("Button", () => {
  it("should render without errors", () => {
    expect(mount(<Button />)).toMatchSnapshot()
  })
})
