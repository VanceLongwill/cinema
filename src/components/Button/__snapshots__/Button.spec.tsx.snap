// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render without errors 1`] = `
<Button>
  <button
    className="Button"
  />
</Button>
`;
